node/core/approval-voting/src/lib.rs

```diff
@@ -1366,8 +1370,11 @@ async fn handle_approved_ancestor(
 		},
 	);
 
+	let all_approved_max = all_approved_max.map(|(hash, block_number)| {
+		(hash, block_number, candidates)
```

Comment:
I think we've got to do `candidates.reverse()` here, since the dispute coordinator expects them in ascending order and we've been `push`ing in descending order.